heron/schedulers/tests/java/org/apache/heron/scheduler/kubernetes/V1ControllerTest.java

```diff
@@ -737,4 +746,284 @@ public void testConfigureTolerations() {
+    final String volumeNameOne = "VolumeNameONE";
+    final String volumeNameTwo = "VolumeNameTWO";
+    final String claimNameOne = "claim-name-one";
+    final String claimNameTwo = "OnDemand";
+    final String mountPathOne = "/mount/path/ONE";
+    final String mountPathTwo = "/mount/path/TWO";
+    final String mountSubPathTwo = "/mount/sub/path/TWO";
+    Map<String, Map<VolumeClaimTemplateConfigKeys, String>> mapOfOpts =
+        ImmutableMap.of(
+            volumeNameOne, ImmutableMap.of(
+                VolumeClaimTemplateConfigKeys.claimName, claimNameOne,
+                VolumeClaimTemplateConfigKeys.path, mountPathOne),
+            volumeNameTwo, ImmutableMap.of(
+                VolumeClaimTemplateConfigKeys.claimName, claimNameTwo,
+                VolumeClaimTemplateConfigKeys.path, mountPathTwo,
+                VolumeClaimTemplateConfigKeys.subPath, mountSubPathTwo)
+        );
+    final V1Volume volumeOne = new V1VolumeBuilder()
+        .withName(volumeNameOne)
+        .withNewPersistentVolumeClaim()
+          .withClaimName(claimNameOne)
+        .endPersistentVolumeClaim()
+        .build();
+    final V1Volume volumeTwo = new V1VolumeBuilder()
+        .withName(volumeNameTwo)
+        .withNewPersistentVolumeClaim()
+          .withClaimName(claimNameTwo)
+        .endPersistentVolumeClaim()
+        .build();
+    final V1VolumeMount volumeMountOne = new V1VolumeMountBuilder()
+        .withName(volumeNameOne)
+        .withMountPath(mountPathOne)
+        .build();
+    final V1VolumeMount volumeMountTwo = new V1VolumeMountBuilder()
+        .withName(volumeNameTwo)
+        .withMountPath(mountPathTwo)
+        .withSubPath(mountSubPathTwo)
+        .build();
+
+    // Test case container.
+    final List<TestTuple<Pair<List<V1Volume>, List<V1VolumeMount>>,
+        Pair<List<V1Volume>, List<V1VolumeMount>>>> testCases = new LinkedList<>();
+
+    // Default case: No PVC provided.
+    final Pair<List<V1Volume>, List<V1VolumeMount>> actualEmpty =
+        v1ControllerPodTemplate.createPersistentVolumeClaimVolumesAndMounts(new HashMap<>());
+    testCases.add(new TestTuple<>("Generated an empty list of Volumes", actualEmpty,
+        new Pair<>(new LinkedList<>(), new LinkedList<>())));
+
+    // PVC Provided.
+    final Pair<List<V1Volume>, List<V1VolumeMount>> expectedFull =
+        new Pair<>(
+            new LinkedList<>(Arrays.asList(volumeOne, volumeTwo)),
+            new LinkedList<>(Arrays.asList(volumeMountOne, volumeMountTwo)));
+    final Pair<List<V1Volume>, List<V1VolumeMount>> actualFull =
+        v1ControllerPodTemplate.createPersistentVolumeClaimVolumesAndMounts(mapOfOpts);
+    testCases.add(new TestTuple<>("Generated a list of Volumes", actualFull,
+        new Pair<>(expectedFull.first, expectedFull.second)));
+
+    // Testing loop.
+    for (TestTuple<Pair<List<V1Volume>, List<V1VolumeMount>>,
+            Pair<List<V1Volume>, List<V1VolumeMount>>> testCase : testCases) {
```

Comment:
Strange CI erroir because this line is indented 13.

```bash
[ERROR] /home/travis/.cache/bazel/_bazel_travis/be6dac4936703c7eedcb4f5cf38cdd65/execroot/org_apache_heron/heron/schedulers/tests/java/org/apache/heron/scheduler/kubernetes/V1ControllerTest.java:902: 'Pair' have incorrect indentation level 12, expected level should be 13. [Indentation]
```